docs/API/core/app-actions-v2.md

```diff
@@ -0,0 +1,460 @@
+- Send weekly reminders for users to take update their working status
+- Integration with 3rd party tools
+- Return data based on user's rights
+
+---
+
+## Data models and key concepts
+
+1. An app action consists in a unique `name`, an optional `frequency`, `timezone`, `environment` and `triggers`.
+2. An app action has a pipeline of `functions` (2nd gen).
+3. An app action can be created as **scheduled** (when using the `frequency` parameter) or to be run **on-demand**. Additionally, our 2nd generation actions can be triggered by external events (e.g. a data source being updated). This can be configured via the `triggers` property.
+4. An app action can contain a pipeline of functions to run either locally (on device) or remote, or target an app screen to run in the cloud. A result can be given back by the screen both when running on a schedule and when on-demand.
+5. An app action (2nd gen) can run on server when `environment` property is set as `server` or `any`
+6. An app action (2nd gen) can run on client side when `environment` property is set as `client` or `any`
+7. An app action is **limited to 120 seconds of execution time**. After 120 seconds, the action will be killed and a specific timeout error will be returned and saved in the logs.
+8. The payload for on-demand actions is **limited to 2048 characters**.
+9. The result sent from an on-demand app action is **limited to 6MB**.
+10. Only JavaScript assets are loaded when the screen runs as an app action. Assets such as CSS and images will be ignored by the system.
+11. Scheduled app actions will only run the published version of a screen, whereas on-demand actions will run the version from the same environment they are fired from (e.g. Fliplet Viewer, Live apps  )
+
+---
+
+## Configuring function pipeline for app actions
+
+Our latest iteration of app actions require you to define a pipeline of functions to execute when the action runs. Additionally, functions can be nested into sub-pipelines.
+
+First, you want to fetch the array of functions available in the system:
+
+```js
+Fliplet.App.Tasks.Functions.get().then(function (listOfFunctions) {
+  // ...
+});
+```
+
+Each function has the following properties:
+
+- `id`: the Fliplet ID (formerly `widgetId`) of the function
+- `name`: the display name of the function
+- `package`: the package name of the function
+- `description`: the description of the function
+
+Here'a a sample code that creates a new app action with a pipeline of function.
+
+In the example, function `com.fliplet.function.if` function that behaves like an [if statement](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/if...else) and evaluates the contents of the input payload and checks if `foo` is `true`. If so, it will execute function `com.example.function.return-string` with the given settings. If `foo` is `false`, the function will be skipped.
+
+```js
+Fliplet.App.Actions.create({
```

Comment:
This example is missing a trigger, environment etc

Reply:
@debankurghosh2061 Updated accordingly